lib/iris/tests/unit/cube/test_Cube.py

```diff
@@ -2517,6 +2517,30 @@ def test_dim_aux_coords(self):
             ],
         )
 
+    def test_aux_coords_ordering(self):
```

Comment:
Would you need the same for `derived_coords`?

Reply:
Having discussed with @pp-mo, we've agreed that all we need to do is convert this test into a _general_ confirmation of correct sorting, by:
* Changing the test name
* Adding a few `dim_coord`s